pilot/pkg/model/config.go

```diff
@@ -366,6 +366,17 @@ var (
 		Collection:  metadata.ServiceEntry.Collection.String(),
 	}
 
+	// ServiceEntry describes service entries
+	SyntheticServiceEntry = ProtoSchema{
```

Comment:
I'm not sure we need to add another type for these - they're still just ServiceEntry objects, no? If we need to differentiate, we can do it based on a custom label. We're planning on having galley add a label to denote that a ServiceEntry was synthesized from k8s resources.

Reply:
Yeah, I wasn't too sure how we are going to handle synthetic serviceEntries, I was under assumption that we are going to rely on the collection which would require us to declared a new config type e.g `SyntheticServiceEntry`. However labels should totally work fine. I will go ahead and update the logic to look into labels.

Reply:
@nmittler have we established what that label is?

Reply:
In my PR (https://github.com/istio/istio/pull/11293/files#diff-4127bafee0667fc94272addef511d697R19) I'm currently going with the annotation: "networking.istio.io/syntheticResource"